Modules/RatMama.py

```diff
@@ -0,0 +1,178 @@
+    \(                                   # The "(" of "(en-US)"
+    (?P<language_code>.+?)               # "en"
+    (?:                                  # Optional group
+        -(?P<language_country>.+)        # "-", "US" (currently unused)
+    )?                                   # Actually make the group optional.
+    \)                                   # The ")" of "(en-US)"
+    )                                    # End of full language text
+    (?:                                  # Possibly match IRC nickname
+    \s+-\s+                              #  -
+    IRC\s+Nickname:\s*(?P<nick>[^\s]+)   # IRC nickname
+    )?                                   # ... emphasis on "Possibly"
+    )                                    # End of the main capture group
+    \s*                                  # Handle any possible trailing whitespace
+    $                                    # End of pattern
+""")
+
+
+@rule(r"^\s*Incoming Client:", case_sensitive=False, full_message=True, prefixless=True,
+      pass_match=False)
+async def handle_ratmama_announcement(ctx: Context):
+    if ctx.user.nickname.casefold() not in ('ratmama[bot]',):
+        return
+
+    message: str = ctx.words_eol[0]
+    result = re.fullmatch(ratmama_regex, message)
+    client_name: str = result.group("cmdr")
+    system_name: str = result.group("system")
+    platform_name: str = result.group("platform")
+    o2_status: bool = result.group("o2") == "OK"
+    lang_code: str = result.group("language_code")
+    nickname: Optional[str] = result.group("nick")
+
+    exist_rescue: Rescue = board.find_by_name(client_name)
+
+    if exist_rescue:
+        # we got a case already!
+        await ctx.reply(f"{client_name} has reconnected! #{exist_rescue.board_index}")
+        # now let's make it more visible if stuff changed
+        diff_response: str = ""
+        if system_name.casefold() != exist_rescue.system.casefold():
+            diff_response += f"System changed! Was {exist_rescue.system}"
+
+        if platform_name.casefold() != str(exist_rescue.platform).replace('Platforms.', '').casefold():
+            diff_response += "Platform changed! "
+
+        if o2_status != exist_rescue.code_red:
+            diff_response += "O2 Status changed" if o2_status else\
+                "O2 Status changed, it is now CODE RED"
+
+        if not diff_response == "":
+            await ctx.reply(diff_response)
+
+    else:
+        # no case for that name, we have to make our own
+        rescue: Rescue = Rescue(client=client_name, system=system_name, irc_nickname=nickname,
+                                code_red=o2_status, lang_id=lang_code)
+        if platform_name.casefold() == "pc":
+            rescue.platform = Platforms.PC
+        elif platform_name.casefold() == "pc":
+            rescue.platform = Platforms.PS
+        elif platform_name.casefold() == "xb":
+            rescue.platform = Platforms.XB
```

Comment:
dictionary mapping may be more use here.

Reply:
dictionary mapping on an Enum?